client/landing/jetpack-cloud/sections/backups/rewind-flow/restore.tsx

```diff
@@ -1,15 +1,206 @@
-	return <div>Restore</div>;
+//todo: move to dedicated types file
+interface RewindState {
+	state: string;
+	rewind?: {
+		status: 'queued' | 'running' | 'finished' | 'fail';
+	};
+}
+
+const BackupRestoreFlow: FunctionComponent< Props > = ( { rewindId, siteId } ) => {
+	const dispatch = useDispatch();
+	const moment = useLocalizedMoment();
+	const translate = useTranslate();
+
+	const [ rewindConfig, setRewindConfig ] = useState< RewindConfig >( defaultRewindConfig );
+	const [ userHasRequestedRestore, setUserHasRequestedRestore ] = useState< boolean >( false );
+
+	const siteUrl = useSelector( state => getSiteUrl( state, siteId ) );
+	const rewindState = useSelector( state => getRewindState( state, siteId ) ) as RewindState;
+
+	const loading = rewindState.state === 'uninitialized';
+
+	const restoreTimestamp = moment.unix( rewindId ).format( 'LLL' );
+
+	const inProgressRewindStatus = useSelector( state =>
+		getInProgressRewindStatus( state, siteId, rewindId )
+	);
+	const inProgressRewindPercentComplete = useSelector( state =>
+		getInProgressRewindPercentComplete( state, siteId, rewindId )
+	);
+
+	const requestRestore = useCallback(
+		() => dispatch( rewindRestore( siteId, rewindId, rewindConfig ) ),
+		[ dispatch, rewindConfig, rewindId, siteId ]
+	);
+
+	const onConfirm = () => {
+		setUserHasRequestedRestore( true );
+		requestRestore();
+	};
+
+	const renderConfirm = () => (
+		<>
+			<div className="rewind-flow__header">
+				<Gridicon icon="history" size={ 48 } />
+			</div>
+			<h3 className="rewind-flow__title">{ translate( 'Restore site' ) }</h3>
+			<p className="rewind-flow__info">
+				{ translate(
+					'{{strong}}%(restoreTimestamp)s{{/strong}} is the selected point for your restore. ',
+					{
+						args: {
+							restoreTimestamp,
+						},
+						components: {
+							strong: <strong />,
+						},
+					}
+				) }
+			</p>
+			<h4 className="rewind-flow__cta">{ translate( 'Choose the items you wish to restore:' ) }</h4>
+			<RewindConfigEditor currentConfig={ rewindConfig } onConfigChange={ setRewindConfig } />
+			<RewindFlowNotice
+				gridicon="notice"
+				title={ translate( 'This will override and remove all content after this point' ) }
+				type={ RewindFlowNoticeLevel.WARNING }
+			/>
+			<Button
+				className="rewind-flow__primary-button"
+				primary
+				onClick={ onConfirm }
+				disabled={ Object.values( rewindConfig ).every( setting => ! setting ) }
+			>
+				{ translate( 'Confirm restore' ) }
+			</Button>
+		</>
+	);
+
+	const renderInProgress = () => (
+		<>
+			<div className="rewind-flow__header">
+				<Gridicon icon="history" size={ 48 } />
+			</div>
+			<h3 className="rewind-flow__title">{ translate( 'Currently restoring your site' ) }</h3>
+			<ProgressBar percent={ inProgressRewindPercentComplete } />
+			<p className="rewind-flow__info">
+				{ translate(
+					'We are restoring your site back to {{strong}}%(restoreTimestamp)s{{/strong}}.',
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **4** times:
`translate( 'We are restoring your site back to {{strong}}%(restoreTimestamp)s{{/strong}}' )` *ES Score: 9*